plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -16,36 +16,54 @@
 package com.google.cloud.tools.eclipse.appengine.login;
 
 import com.google.api.client.auth.oauth2.Credential;
-import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeTokenRequest;
-import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
-import com.google.api.client.googleapis.util.Utils;
-import com.google.cloud.tools.eclipse.appengine.login.ui.GoogleLoginBrowser;
+import com.google.cloud.tools.eclipse.appengine.login.ui.LoginServiceUi;
+import com.google.cloud.tools.ide.login.GoogleLoginState;
+import com.google.cloud.tools.ide.login.LoggerFacade;
+import com.google.cloud.tools.ide.login.OAuthDataStore;
+import com.google.cloud.tools.ide.login.UiFacade;
+import com.google.common.annotations.VisibleForTesting;
 
-import org.eclipse.e4.core.contexts.IEclipseContext;
-import org.eclipse.jface.window.IShellProvider;
-import org.eclipse.ui.PlatformUI;
-
-import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * Provides service related to login, e.g., account management, getting a credential of a
  * currently active user, etc.
  */
-public class GoogleLoginService {
-
-  private static final String STASH_OAUTH_CRED_KEY = "OAUTH_CRED";
+public final class GoogleLoginService {
 
   // For the detailed info about each scope, see
   // https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/wiki/Cloud-Tools-for-Eclipse-Technical-Design#oauth-20-scopes-requested
-  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
-  ));
+  private static final SortedSet<String> OAUTH_SCOPES = Collections.unmodifiableSortedSet(
+      new TreeSet<>(Arrays.asList(
+          "email", //$NON-NLS-1$
```

Comment:
that is...odd.  Perhaps they've finally sorted this out. I'll check with
the cloud SDK team.

On Tue, Aug 2, 2016 at 1:32 PM, Chanseok Oh <notifications@github.com>
wrote:

> In
> plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/486#discussion_r73200337>
> :
>
> >
> >    // For the detailed info about each scope, see
> >    // https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/wiki/Cloud-Tools-for-Eclipse-Technical-Design#oauth-20-scopes-requested
> > -  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
> > -      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
> > -  ));
> > +  private static final SortedSet<String> OAUTH_SCOPES = Collections.unmodifiableSortedSet(
> > +      new TreeSet<>(Arrays.asList(
> > +          "email", //$NON-NLS-1$
>
> So far, we didn't need it for standard app engine deployment.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/486/files/9d783a4a3b65898b0709d531ea4d93bfe87480b0#r73200337>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HWpYbjQMRoDvlt55YRW2EGeK4V57ks5qb38wgaJpZM4JakbD>
> .
>